src/wp-includes/option.php

```diff
@@ -253,6 +253,88 @@ function get_option( $option, $default_value = false ) {
 	return apply_filters( "option_{$option}", maybe_unserialize( $value ), $option );
 }
 
+/**
+ * Primes specific options into the cache with a single database query.
+ *
+ * Only options that do not already exist in cache will be primed.
+ *
+ * @since 6.4.0
+ *
+ * @global wpdb $wpdb WordPress database abstraction object.
+ *
+ * @param array $options An array of option names to be primed.
+ */
+function prime_options( $options ) {
+	$alloptions     = wp_load_alloptions();
+	$cached_options = wp_cache_get_multiple( $options, 'options' );
+
+	// Filter options that are not in the cache.
+	$options_to_prime = array();
+	foreach ( $options as $option ) {
+		if ( ! isset( $cached_options[ $option ] ) && ! isset( $alloptions[ $option ] ) ) {
+			$options_to_prime[] = $option;
+		}
+	}
+
+	if ( ! empty( $options_to_prime ) ) {
+		global $wpdb;
+		$option_names = implode( "','", array_map( 'esc_sql', $options_to_prime ) );
+		$query   = "SELECT option_name, option_value FROM {$wpdb->options} WHERE option_name IN ('$option_names')";
```

Comment:
This should use `$wpdb->prepare`, like this:
```suggestion
		$query   = $wpdb->prepare( "SELECT option_name, option_value FROM $wpdb->options WHERE option_name IN %s", $option_names );
```